neuraxle/pipeline.py

```diff
@@ -70,8 +60,23 @@ class Pipeline(BasePipeline, ResumableStepMixin):
     Fits and transform steps after latest checkpoint
     """
 
-    def __init__(self, steps: NamedTupleList):
-        super().__init__(steps)
+    def __init__(self, steps: NamedTupleList, hasher=HasherByIndex()):
+        if hasher is None:
+            raise ValueError('Pipeline hasher cannot be None')
+
+        BasePipeline.__init__(self, steps=steps, hasher=hasher)
+
+    def transform(self, data_inputs: Any):
+        """
+        After loading the last checkpoint, transform each pipeline steps
+        :param data_inputs: the data input to transform
+        :return: transformed data inputs
+        """
+        ids = self.hasher.hash(self.hyperparams, data_inputs)
+        data_container = DataContainer(ids=ids, data_inputs=data_inputs)
```

Comment:
When do we set `ids_rehashed = ids` at the beginning once? Should we initialize `ids_rehashed` at the same time of `ids` like I suggest?

Reply:
Right now, **id_rehashed** is named as **ids** in the **DataContainer** class. DataContainer has original_ids, and ids in it